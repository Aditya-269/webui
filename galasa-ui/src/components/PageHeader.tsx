/*
 * Copyright contributors to the Galasa project
 */
'use client';

import { Header, HeaderName , SkipToContent} from '@carbon/react';

export default function PageHeader() {
  return (
    <Header aria-label="GALASA">
      <SkipToContent />
      <HeaderName prefix="Galasa Ecosystem">(Experimental)</HeaderName>
    </Header>
  );
};