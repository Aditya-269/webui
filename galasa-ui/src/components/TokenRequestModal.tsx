 /*
 * Copyright contributors to the Galasa project
 */
'use client';

import { Button, Modal } from '@carbon/react';
import { useState } from 'react';
import { TextInput, PasswordInput } from '@carbon/react';
import { TokenTable } from './Table';
import { InlineNotification } from '@carbon/react';

const headers = [
  {
    key: 'tokenName',
    header: 'Token',
  },
  {
    key: 'scope',
    header: 'Scope',
  },
  {
    key: 'expires',
    header: 'Expires',
  },
];

const rows = [
  {
    id: '1234',
    tokenName: 'tkn1Example',
    scope: 'ALL',
    expires: '2023-10-22',
  },
  {
    id: '5678',
    tokenName: 'tkn2Example',
    scope: 'ALL',
    expires: '2023-09-31',
  },
];

export default function TokenRequestModal() {
  const [open, setOpen] = useState(false);
  const [error, setError] = useState('');

  const submitTokenRequest = async () => {
    const name = (document.getElementById('name-txtinput') as HTMLInputElement).value;
    const secret = (document.getElementById('secret-txtinput') as HTMLInputElement).value;
    const codedSecret = Buffer.from(secret).toString('base64');

    // Call out to /auth/token with the payload for the name and secret for dex
    const tokenUrl = '/auth/token';
    const response = await fetch(tokenUrl, {
      method: 'POST',
      body: JSON.stringify({ name, secret: codedSecret }),
      headers: { 'Content-type': 'application/json; charset=UTF-8' },
    });

    const responseJson = await response.json();
    if (responseJson.error) {
      setError(responseJson.error);
    } else {
      // Redirect to authenticate with Dex
      window.location.replace(responseJson.url);
    }
  };
  return (
    <>
      <Button onClick={() => setOpen(true)}>Request Access Token</Button>
      <TokenTable headers={headers} rows={rows} />
      <Modal
        modalHeading="Request a new Personal Access Token"
        modalLabel="Access Tokens"
        primaryButtonText="Submit"
        secondaryButtonText="Cancel"
        open={open}
        onRequestClose={() => {
          setOpen(false);
          setError('');
        }}
        onRequestSubmit={async () => {
          await submitTokenRequest();
        }}
      >
        <TextInput data-modal-primary-focus id="name-txtinput" labelText="Token Name" style={{ marginBottom: '1rem' }} />
        <PasswordInput
          data-modal-primary-focus
          id="secret-txtinput"
          labelText="Secret"
          helperText="The secret you would like to use with this token"
          style={{ marginBottom: '1rem' }}
        />
        {error && (
          <InlineNotification
            title="Error Requesting Access Token"
            subtitle={error}
            kind="error"
            onCloseButtonClick={() => setError('')}
            lowContrast
          />
        )}
      </Modal>
    </>
  );
};
