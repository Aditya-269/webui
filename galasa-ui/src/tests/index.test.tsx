/*
 * Copyright contributors to the Galasa project
 */
import TokenRequestModal from '@/components/Modals';
import PageHeader from '@/components/pageHeader';
import { render, screen } from '@testing-library/react';

test('renders Galasa Ecosystem header', () => {
  render(<PageHeader />);
  const titleElement = screen.getByText(/Galasa Ecosystem/i);
  expect(titleElement).toBeInTheDocument();
});

test('renders Galasa Modal Token Request', () => {
  render(<TokenRequestModal />);
  const buttonElement = screen.getByText(/Request Access Token/i);
  const requestModalElement = screen.getByText(/Request a new Personal Access Token/i);
  const responseModalElement = screen.getByText(/Your new access token is:/i);
  expect(buttonElement).toBeInTheDocument();
  expect(requestModalElement).toBeInTheDocument();
  expect(responseModalElement).toBeInTheDocument();
});
