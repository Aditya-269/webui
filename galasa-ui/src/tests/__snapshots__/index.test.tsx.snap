// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Galasa Ecosystem homepage 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="content"
      >
        <div
          class="cds--tile"
          id="tile"
        >
          Home
        </div>
        <section>
          <div
            class="homeContentWrapper"
          >
            <section>
              <h2>
                Welcome to your Galasa Service
              </h2>
              <section>
                <h4>
                  Get the most from your Galasa experience by reading the
                  <a
                    class="link"
                    href="https://galasa.dev/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                     Galasa documentation.
                  </a>
                </h4>
              </section>
            </section>
            <img
              alt="home-graphic"
              class="heroImage"
              data-nimg="1"
              decoding="async"
              height="680"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=1920&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=750&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1920&q=75 2x"
              style="color: transparent;"
              width="680"
            />
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="content"
    >
      <div
        class="cds--tile"
        id="tile"
      >
        Home
      </div>
      <section>
        <div
          class="homeContentWrapper"
        >
          <section>
            <h2>
              Welcome to your Galasa Service
            </h2>
            <section>
              <h4>
                Get the most from your Galasa experience by reading the
                <a
                  class="link"
                  href="https://galasa.dev/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                   Galasa documentation.
                </a>
              </h4>
            </section>
          </section>
          <img
            alt="home-graphic"
            class="heroImage"
            data-nimg="1"
            decoding="async"
            height="680"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=1920&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=750&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1920&q=75 2x"
            style="color: transparent;"
            width="680"
          />
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
